doc/Language/traits.pod6

```diff
@@ -108,6 +108,27 @@ would be a way of adding functionality which cuts across class and role
 hierarchies, or can even be used to add behaviors to independently defined
 routines.
 
+=head2 X<C<is implementation-detail>,traits> trait
+
+Note: this trait is available in Rakudo compiler starting from 2020.05 release.
+
+The C<is> with C<implementation-detail> argument is used by Raku
+language implementations and module authors to mark particular
+routines (including methods) as not meant to be a public API.
```

Comment:
"part of", maybe. Also clarify what's exactly meant by this, that is, that you can use them and they are public-facing, but they can be removed or changed at any time in the future, so future-proof related code.